blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/listeners/NodeDownstreamBuildAction.java

```diff
@@ -57,4 +70,41 @@ public boolean equals(Object o) {
     public int hashCode() {
         return Objects.hash(link, description);
     }
+
+    @Extension
+    public static class FactoryImpl extends TransientActionFactory<FlowNode> {
+
+        @Override
+        public Class<FlowNode> type() {
+            return FlowNode.class;
+        }
+
+        @Override
+        public Collection<? extends Action> createFor(FlowNode node) {
+            try {
+                Queue.Executable executable = node.getExecution().getOwner().getExecutable();
+                if (executable instanceof Actionable) {
+                    DownstreamBuildAction action = ((Actionable) executable).getAction(DownstreamBuildAction.class);
+                    if (action != null) {
+                        for (DownstreamBuildAction.DownstreamBuild downstreamBuild : action.getDownstreamBuilds()) {
+                            if (downstreamBuild.getFlowNodeId().equals(node.getId())) {
+                                Run<?, ?> downstream = downstreamBuild.getBuild();
```

Comment:
Everything before this is probably ok for a `TransientActionFactory`, but in the worst case, this line will have to load the downstream build, which may be slow, so IDK if we are ok with this approach.

Another option could be to keep `DownstreamJobListener` and just change the log level of the warning message to avoid log spam. `NodeDownstreamBuildAction` would not work in cases where the upstream build completes before the downstream build starts, but maybe that's ok. 

A third option could be to delete `NodeDownstreamBuildAction` and switch to using `DownstreamBuildAction` directly, but from a brief look, that seems like it would be a more involved change, and I think it might have similar performance concerns as the `TransientActionFactory` (not 100% sure though).